examples/sqlite/relations/src/main.rs

```diff
@@ -0,0 +1,206 @@
+use diesel::prelude::*;
```

Comment:
All these files look fine - I diffed them against the postgres versions and only the expected changes were present.

Not for this PR, but more generally - is there a way we can do better, maybe with macros or preprocessing, so that we can have all the code written once, and compilation is attempted against all the backends?

Reply:
It's quite hard to do that with macros as there are some subtle differences between each of the backends in terms of supported SQL features. There is the `#[derive(MultiConnection)]` macro that allows you to write backend "agnostic" code, but these examples should demonstrate that you can in fact optimize your code for a specific backend (and use features that are only supported by this backend).

Reply:
OK, makes sense. I'll keep pondering this, though.